packages/openactive-integration-tests/test/features/approval/minimal-proposal/implemented/accept-proposal-book-test.js

```diff
@@ -31,101 +38,90 @@ FeatureHelper.describeFeature(module, {
 
-  describe('Get Opportunity Feed Items', () => {
-    (new GetMatch({
-      state, flow, logger, orderItemCriteria,
-    }))
-      .beforeSetup()
-      .successChecks()
-      .validationTests();
+  // ## Initiate Flow Stages
+  const fetchOpportunities = FetchOpportunitiesFlowStage.create({
+    orderItemCriteriaList,
+    logger,
+    requestHelper,
   });
-
-  describe('C1', () => {
-    (new C1({
-      state, flow, logger,
-    }))
-      .beforeSetup()
-      .successChecks()
-      .validationTests();
-
-    itShouldReturnOrderRequiresApprovalTrue(() => state.c1Response);
+  const c1 = C1FlowStage.create({
+    prerequisite: fetchOpportunities,
+    logger,
+    requestHelper,
   });
-
-  describe('C2', () => {
-    (new C2({
-      state, flow, logger,
-    }))
-      .beforeSetup()
-      .successChecks()
-      .validationTests();
-
-    itShouldReturnOrderRequiresApprovalTrue(() => state.c2Response);
+  const c2 = C2FlowStage.create({
+    prerequisite: c1,
+    logger,
+    requestHelper,
   });
-
-  describe('P', () => {
-    (new P({
-      state, flow, logger,
-    }))
-      .beforeSetup()
-      .successChecks()
-      .validationTests();
-
-    // TODO does validator check that orderProposalVersion is of form {orderId}/versions/{versionUuid}
-    it('should include an orderProposalVersion, of the form {orderId}/versions/{versionUuid}', () => {
-      expect(state.pResponse.body).to.have.property('orderProposalVersion')
-        .which.matches(RegExp(`${state.uuid}/versions/.+`));
-    });
-    // TODO does validator check that orderItemStatus is https://openactive.io/OrderItemProposed
-    // TODO does validator check that full Seller details are included in the seller response?
+  const p = PFlowStage.create({
+    prerequisite: c2,
+    logger,
+    requestHelper,
   });
-
-  describe('Simulate Seller Approval (Test Interface Action)', () => {
-    (new TestInterfaceAction({
-      flow,
-      logger,
-      createActionFn: () => ({
-        type: 'test:SellerAcceptOrderProposalSimulateAction',
-        objectType: 'OrderProposal',
-        objectId: state.pResponse.body['@id'],
-      }),
-      completedFlowStage: 'P',
-    }))
-      .beforeSetup()
-      .successChecks();
+  const initiateOrderFeedUpdate = OrderFeedUpdateFlowStage.createInitiator({
+    prerequisite: p,
+    requestHelper,
+  });
+  const simulateSellerApproval = TestInterfaceActionFlowStage.create({
+    testName: 'Simulate Seller Approval (Test Interface Action)',
+    prerequisite: initiateOrderFeedUpdate,
+    createActionFn: () => ({
+      type: 'test:SellerAcceptOrderProposalSimulateAction',
+      objectType: 'OrderProposal',
+      objectId: p.getResponse().body['@id'],
+    }),
+    requestHelper,
+  });
+  const collectOrderFeedUpdate = OrderFeedUpdateFlowStage.createCollector({
+    testName: 'Order Feed Update (after Simulate Seller Approval)',
+    prerequisite: simulateSellerApproval,
+    logger,
+  });
```

Comment:
See the documentation for OrderFeedUpdateFlowStage to understand what these 2 "initiate" and "collect" stages are all about.